### PR TITLE
SUS-399: Deleting CSS files cannot be cleanly done in production

### PR DESCRIPTION
**Pull request by mszabo-wikia**
ticket & CR notes: https://wikia-inc.atlassian.net/browse/SUS-399

@macbre @mixth-sense

### CODE REVIEW

File: extensions/wikia/AssetsManager/builders/AssetsManagerSassBuilder.class.php [hunk 1]
```diff
@@ -4,40 +4,54 @@
...
 	public function getContent( $processingTimeStart = null ) {
 		global $IP, $wgEnableSASSSourceMaps;
-		wfProfileIn(__METHOD__);
+		wfProfileIn( __METHOD__ );
+
+		// SUS-399: Abort early if we received a request for a file that no longer exists
+		if ( !$this->exists ) {
+			wfProfileOut( __METHOD__ );
+			return '';
+		}
```
Comment (macbre) [inline extensions/wikia/AssetsManager/builders/AssetsManagerSassBuilder.class.php:71]: Will such requests still be handled correctly - i.e. return [`HTTP 404`](http://slot1.images4.wikia.nocookie.net/__am/1469493438/sasses/background-dynamic%3Dfalse%26background-image%3Dhttp%253A%252F%252Fvignette3.wikia.nocookie.net%252Fpoznan%252Fimages%252F5%252F50%252FWiki-background%252Frevision%252Flatest%253Fcb%253D20150517124735%2526path-prefix%253Dpl%26background-image-height%3D288%26background-image-width%3D300%26color-body%3D%2523f3e2d0%26color-body-middle%3D%2523f3e2d0%26color-buttons%3D%2523006cb0%26color-header%3D%25233a5766%26color-links%3D%2523006cb0%26color-page%3D%2523f8f4f0%26oasisTypography%3D1%26page-opacity%3D92%26widthType%3D0/skins/oasis/css/oasis.scss,extensions/wikia/Recirculation/styles/recirculation.scss,extensions/wikia/PortableInfobox/styles/PortableInfobox.scss,extensions/wikia/VisualEditorTourExperiment/styles/VisualEditorTourExperimentInit.scss,extensions/wikia/GlobalShortcuts/styles/GlobalShortcutsHelp.scss,extensions/wikia/GlobalShortcuts/styles/GlobalShortcutsSearch.scss,extensions/wikia/SpecialPromote/css/SpecialPromote.scss,extensions/wikia/GlobalNavigation/styles/GlobalNavigation.scss,extensions/wikia/GlobalNavigation/styles/GlobalNavigationSearch.scss,extensions/wikia/GlobalNavigation/styles/GlobalNavigationAccountNavigation.scss,extensions/wikia/GlobalNavigation/styles/GlobalNavigationUserLoginDropdown.scss,extensions/wikia/GlobalNavigation/styles/GlobalNavigationNotifications.scss,extensions/wikia/GlobalNavigation/styles/GlobalNavigationInverse.scss,extensions/wikia/GlobalNavigation/styles/GoogleSearchAPI.scss,skins/shared/styles/transparent-out.scss,extensions/wikia/PageShare/styles/PageShare.scss,extensions/wikia/GlobalFooter/styles/GlobalFooter.scss,extensions/wikia/Qualaroo/css/Qualaroo.scass)?

Comment (macbre) [inline extensions/wikia/AssetsManager/builders/AssetsManagerSassBuilder.class.php:71]: Scratch the above, we should silently skip such cases. :+1:

Comment (mszabo-wikia) [inline extensions/wikia/AssetsManager/builders/AssetsManagerSassBuilder.class.php:71]: yes - if this is a group request for multiple sasses, this means the rest will load correctly and we'll just load nothing for the deleted file :)

Comment (macbre) [inline extensions/wikia/AssetsManager/builders/AssetsManagerSassBuilder.class.php:71]: Maybe let's at least emit a CSS/JS comment block saying that this file does not exist. To ease debugging.

Comment (mszabo-wikia) [inline extensions/wikia/AssetsManager/builders/AssetsManagerSassBuilder.class.php:71]: done in b35e192

**Comment by macbre** [comment]
:+1:

File: extensions/wikia/AssetsManager/builders/AssetsManagerSassBuilder.class.php [hunk 2]
```diff
@@ -4,40 +4,59 @@
...
 		}
 
-		if (!endsWith($this->mOid, '.scss', false)) {
-			throw new Exception('Requested file must be .scss.');
+		if ( !endsWith( $this->mOid, '.scss', false ) ) {
+			throw new Exception( 'Requested file must be .scss.' );
 		}
 
 		//remove slashes at the beginning of the string, we need a pure relative path to open the file
 		$this->mOid = preg_replace( '/^[\/]+/', '', $this->mOid );
```
Comment (ghost) [inline extensions/wikia/AssetsManager/builders/AssetsManagerSassBuilder.class.php]: This can be easily done with `(l)trim( $this->mOid, '/' )`

**Comment by ghost** [comment]
Just one comment.  Pattern matching is a bit of an overkill when removing trailing slashes is all we need. `ltrim()` will start processing the given string byte by byte starting from the beginning and terminate as soon as the first non-match is reached, whereas `preg_replace()` first need to compute the given pattern, match the given string with the expression and do the replace only when "entire string matches the entire pattern".

**Pull request merged**